Realm/RLMSyncPermission.h

```diff
@@ -16,7 +16,250 @@
+
+    /// If `false`, no modifications to objects of this type are permitted. Write
+    /// transactions modifying the objects can be performed locally, but any changes made
+    /// will be reverted by the server.
+    ///
+    /// Deleting an object is considered a modification, and is governed by this privilege.
+    bool update : 1;
+
+    /// If `false`, the User is not permitted to create new subscriptions for this class.
+    /// Local queries against the objects within the Realm will work, but new
+    /// subscriptions will never add objects to the Realm.
+    bool subscribe : 1;
+
+    /// If `false`, no modifications to the permissions property of the RLMClassPermissions
+    /// object for this type are permitted. Write transactions can be performed locally,
+    /// but any changes made will be reverted by the server.
+    ///
+    /// Note that if invalid privilege changes are made, `-[RLMRealm privilegesFor*:]`
+    /// will return results reflecting those invalid changes until synchronization occurs.
+    ///
+    /// Even if this field is `true`, note that the user will be unable to grant
+    /// privileges to a Role which they do not themselves have.
+    bool setPermissions : 1;
+};
+
+/**
+ A description of the actual privileges which apply to a specific RLMObject.
+
+ This is a combination of all of the privileges granted to all of the Roles which the
+ current User is a member of, obtained by calling `-[RLMRealm privilegesForObject:]` on
+ the Realm.
+
+ By default, all operations are permitted, and each privilege field indicates an operation
+ which may be forbidden.
+ */
+struct RLMObjectPrivileges {
+    /// If `false`, the current User is not permitted to read this object directly.
+    ///
+    /// Objects which cannot be read by a user will appear in a Realm due to that read
+    /// permissions are transitive. All objects which a readable object links to are
+    /// themselves implicitly readable. If the link to an object with `read=false` is
+    /// removed, the object will be deleted from the local Realm.
+    bool read : 1;
+
+    /// If `false`, modifying the fields of this type is not permitted. Write
+    /// transactions modifying the objects can be performed locally, but any changes made
+    /// will be reverted by the server.
+    ///
+    /// Note that even if this is `true`, the user may not be able to modify the
+    /// `RLMArray<RLMPermission> *` property of the object (if it exists), as that is
+    /// governed by `setPermissions`.
+    bool update : 1;
+
+    /// If `false`, deleting this object is not permitted. Write transactions which delete
+    /// the object can be performed locally, but the server will restore it.
+    ///
+    /// It is possible to have `update` but not `delete` privileges, or vice versa. For
+    /// objects with primary keys, `delete` but not `update` is ill-advised, as an object
+    /// can be updated by deleting and recreating it.
+    bool del : 1;
```

Comment:
I assume this is `del` to remain compatible with Objective-C++ where `delete` is a keyword? Something like `remove` or even `delete_` might be clearer?